src/algebra/algebra/spectrum.lean

```diff
@@ -431,4 +431,12 @@ begin
              coe_subset_nonunits (φ.to_ring_hom.ker_ne_top) h],
 end
 
+lemma mem_resolvent_set_apply (φ : A →ₐ[R] B) {a : A} {r : R} (h : r ∈ resolvent_set R a) :
+  r ∈ resolvent_set R (φ a) :=
+by simpa only [to_ring_hom_eq_coe, ring_hom.to_monoid_hom_eq_coe, ring_hom.coe_monoid_hom,
+  coe_to_ring_hom, map_sub, commutes] using h.map φ.to_ring_hom.to_monoid_hom
```

Comment:
Does this work?
```suggestion
  coe_to_ring_hom, map_sub, commutes] using h.map φ
```
Then you can likely remove most of the lemmas in `simp`

Reply:
no, I checked.

Reply:
#12577 ought to make it work

Reply:
If you don't want to wait for #12577, then using `ring_hom.is_unit_map` is fine; while eventually we should remove `ring_hom.is_unit_map`, I have no intention of doing so in #12577.